arroyo/backends/kafka/consumer.py

```diff
@@ -526,21 +526,23 @@ def paused(self) -> Sequence[Partition]:
 
         return [*self.__paused]
 
-    def stage_offsets(self, offsets: Mapping[Partition, int]) -> None:
+    def stage_offsets(self, offsets: Mapping[Partition, Offset]) -> None:
         if self.__state in {KafkaConsumerState.CLOSED, KafkaConsumerState.ERROR}:
             raise InvalidState(self.__state)
 
         if offsets.keys() - self.__offsets.keys():
             raise ConsumerError("cannot stage offsets for unassigned partitions")
 
-        self.__validate_offsets(offsets)
+        self.__validate_offsets(
+            {key: offset.offset for (key, offset) in offsets.items()}
```

Comment:
Nit: Would be easier to read code if you renamed "key" to "partition" wherever Offset is used.